File: 2015/acm_com_03_10_2015/Тесты/parts/src/TestGen.java
import java.io.*;
import java.util.*;

public class TestGen {

    static PrintWriter out;
    static Random rnd = new Random(0);
    static int tNumber = 1;
    
    
    static void printTest(int n, int[] a, int[] b) throws IOException {
        System.out.println(String.format("Generating test %d", tNumber));
        out = new PrintWriter(String.format("../tests/%02d", tNumber++));
        out.println(n);
        for (int i = 0; i < n; i++) {
            out.println(a[i] + " " + b[i]);
        }
        out.close();
    }

    
    static void genSample() throws IOException {
        printTest(4, new int[]{3, 7, 2, 1}, new int[]{5, 9, 4, 10});
    }
    
    static int randFromTo(int l, int r) {
        return rnd.nextInt(r - l + 1) + l;      
    }

    public static void main(String[] args) throws IOException {
        new File("../tests").mkdir();        
        genSample();
        for (int it = 0; it < 5; ++it) {
            int n = randFromTo(3, 20);
            int[] a = new int[n];
            int[] b = new int[n];
            for (int i = 0; i < n; ++i) {
                b[i] = randFromTo(1, 10);
                a[i] = randFromTo(1, b[i]);
            }
            printTest(n, a, b);
        }

        {
            int n = 1;
            int[] a = new int[n];
            int[] b = new int[n];
            a[0] = randFromTo(1, 10000);
            b[0] = a[0];
            printTest(n, a, b);
        }
        
        {
            int n = 1;
            int[] a = new int[n];
            int[] b = new int[n];
            b[0] = randFromTo(1, 10000);
            a[0] = randFromTo(1, b[0]);
            printTest(n, a, b);
        }

        {
            int n = 10;
            int[] a = new int[n];
            int[] b = new int[n];
            for (int i = 0; i < n; ++i) {
                a[i] = randFromTo(1, 10000);
                b[i] = a[i];
            }
            printTest(n, a, b);
        }

        for (int it = 0; it < 10; ++it) {
            int n = randFromTo(10000, 100000);
            int[] a = new int[n];
            int[] b = new int[n];
            for (int i = 0; i < n-1; ++i) {
                b[i] = randFromTo(1, 1000000000);
                a[i] = randFromTo(b[i]/2, b[i]);
            }
            a[n-1] = randFromTo(1, 1000000000);
            b[n-1] = a[n-1];
            printTest(n, a, b);
        }
        
        {
            int n = 100000;
            int[] a = new int[n];
            int[] b = new int[n];
            for (int i = 0; i < n; ++i) {
                a[i] = 1;
                b[i] = 1000000000;
            }
            printTest(n, a, b);
        }
        
        {
            int n = 100000;
            int[] a = new int[n];
            int[] b = new int[n];
            for (int i = 0; i < n; ++i) {
                a[i] = 1000000000-1;
                b[i] = 1000000000;
            }
            printTest(n, a, b);
        }
        
        {
            int n = 100000;
            int[] a = new int[n];
            int[] b = new int[n];
            for (int i = 0; i < n; ++i) {
                b[i] = randFromTo(1, 1000000000);
                a[i] = Math.max(1, b[i] - 1);
            }
            printTest(n, a, b);
        }
        
        for (int it = 0; it < 5; ++it) {
            int n = randFromTo(10000, 100000);
            int[] a = new int[n];
            int[] b = new int[n];
            for (int i = 0; i < n; ++i) {
                b[i] = randFromTo(100000000, 1000000000);
                a[i] = randFromTo(1, b[i]);
            }
            printTest(n, a, b);
        }

    }                 
}

File: 2015/acm_com_03_10_2015/Тесты/cinema/src/TestGen.py
import random
random.seed("basic01/cinema")

class TestGen:
    def __init__(self):
        self.test_number = 0

    def generate_samples(self):
        self.print_test(3, 1, 1)
        self.print_test(3, 1, 2)
        
    def generate_min_tests(self):
        self.print_test(1, 1, 1)
        self.print_test(1, 1, 2)
        self.print_test(1, 2, 1)
        self.print_test(1, 1, 1000)
        self.print_test(3, 1000, 1)
        self.print_test(5, 1, 1)
        self.print_test(5, 1, 2)
        self.print_test(5, 2, 1)
        self.print_test(5, 2, 2)
        self.print_test(5, 2, 3)
        self.print_test(5, 3, 2)
        self.print_test(5, 3, 3)
                                                                    
    def generate_hand_tests(self):
        pass

    def generate_random_test(self, maxn):
        n = random.randint(1, (maxn - 1) // 2 + 1) * 2 + 1
        a = random.randint(1, n // 2)
        b = random.randint(1, n // 2)
        if random.randint(1, 2) == 1:
            if random.randint(1, 2) == 1:
                a = n // 2 + 1
            if random.randint(1, 2) == 1:
                b = n // 2 + 1
        self.print_test(n, a, b)                

    def generate_max_tests(self, maxn):
        n = maxn
        if n % 2 == 0:
            n -= 1
        a = n // 2
        b = n // 2
        self.print_test(n, a, b)
        self.print_test(n, a - random.randint(1, 5), b)
        self.print_test(n, a, b - random.randint(1, 5))
        self.print_test(n, a - random.randint(1, 5), b - random.randint(1, 5))

    def print_test(self, n, a, b):
        self.test_number += 1
        print('Generating test #%d' % self.test_number)

        test_name = "../tests/" + "{0:0=2d}".format(self.test_number)
        test_file = open(test_name, "w")
        print(n, a, b, file=test_file)
        test_file.close()

    def generate_all_tests(self):
        self.generate_samples()
        self.generate_min_tests()
        self.generate_hand_tests()
        
        for i in range(5):
            self.generate_random_test(50)
        for i in range(5):
            self.generate_random_test(300)
        for i in range(5):
            self.generate_random_test(1000)
        self.generate_max_tests(1000)

import os, shutil
if os.path.exists('../tests'):
    print('Directory tests already exists')
    shutil.rmtree('../tests', ignore_errors=False, onerror=None)
try:
    os.makedirs('../tests')
except OSError:
    pass

import time
start = time.time()
writer = TestGen()
writer.generate_all_tests()
finish = time.time()
print('All tests were generated. Elapsed time: ', finish - start)
